package/system/gpio-cdev/nu801/Makefile

```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=nu801
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/chunkeey/nu801.git
+PKG_SOURCE_VERSION:=794a588fce3150129bb2e5bb37590c7daceb8927
+
+PKG_MAINTAINER:=Christian Lamparter <chunkeey@gmail.com>
+PKG_LICENSE:=GPL-3.0-or-later
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/nu801
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=Userspace GPIO Drivers
+  DEPENDS:=@!LINUX_5_4 +kmod-leds-uleds
+  KCONFIG:= \
+    CONFIG_GPIO_CDEV=y
```

Comment:
I do not like when normal packages activate a Kconfig option. I am not even sure if this works correctly,

Reply:
Would you prefer the generic kernel 5.10 kernel config enables this then? I'm open to changing how this is done, but I want to make sure it's fixed correctly.

Reply:
for what's it worth: The KCONFIG:= variable is declared in include/package-defaults.mk's ````Package/Default```` section.  which means that every package (doesn't matter if kernel or not) can set it.

Now, about that "not sure if it works". I think It does. (with the caveat that it works in the same way as any other kmod-package. (i.e. for it to work, it has to be included during build as either CONFIG_PACKAGE_NU801=m or =y), so the
requested `CONFIG_GPIO_CDEV` is set. This is done by:

https://github.com/openwrt/openwrt/blob/master/include/kernel-defaults.mk#L112

> [...]package-metadata.pl kconfig $(TMP_DIR)/.packageinfo $(TOPDIR)/.config $(KERNEL_PATCHVER) > $(LINUX_DIR)/.config.override

the package-metadata.pl script which generates the ````.config.override````file which gets passed to the final kernel
(sync?)config step takes that dependency from the same $(TMP_DIR)/.packageinfo as with every other package.